sdk/src/main/java/io/opentelemetry/sdk/internal/ComponentRegistry.java

```diff
@@ -53,22 +53,14 @@ public V get(String instrumentationName) {
   public V get(String instrumentationName, @Nullable String instrumentationVersion) {
     InstrumentationLibraryInfo instrumentationLibraryInfo =
         InstrumentationLibraryInfo.create(instrumentationName, instrumentationVersion);
-    V tracer = registry.get(instrumentationLibraryInfo);
-    if (tracer == null) {
-      synchronized (lock) {
-        // Re-check if the value was added since the previous check, this can happen if multiple
-        // threads try to access the same named tracer during the same time. This way we ensure that
-        // we create only one TracerSdk per name.
-        tracer = registry.get(instrumentationLibraryInfo);
-        if (tracer != null) {
-          // A different thread already added the named Tracer, just reuse.
-          return tracer;
-        }
-        tracer = newComponent(instrumentationLibraryInfo);
-        registry.put(instrumentationLibraryInfo, tracer);
-      }
+
+    V component = registry.get(instrumentationLibraryInfo);
+    if (component != null) {
+      return component;
     }
-    return tracer;
+
+    registry.putIfAbsent(instrumentationLibraryInfo, newComponent(instrumentationLibraryInfo));
+    return registry.get(instrumentationLibraryInfo);
```

Comment:
Should we add a test for this? (Since the previous line returned the wrong value, etc) ;)

Reply:
Not sure what you want, the correct return value is tested, if you want to test that putIfAbsent is called twice we need a newComponent implementation that blocks first time then second time is ok, then check that returns the same instance in the end. We still create 2 instances but one we drop immediately which may be something worth documented.

Reply:
Oh, never mind, I thought the `return registry.put(instrumentationLibraryInfo, tracer);` line had been kept.

Reply:
yeah, testing this would be extraordinarily tricky, since it would depend on very finicky timing, and I'm not even sure how you would assert anything useful out of it. :)